site/en/tutorials/load_data/tf-records.ipynb

```diff
@@ -325,190 +415,216 @@
       ]
     },
     {
       "metadata": {
-        "id": "MKPHzoGv7q44",
-        "colab_type": "code",
-        "colab": {}
+        "id": "GmehkCCT81Ez",
+        "colab_type": "text"
       },
-      "cell_type": "code",
+      "cell_type": "markdown",
       "source": [
-        "# Write the tf.Example observations to test.tfrecords.\n",
-        "\n",
-        "writer = tf.python_io.TFRecordWriter('test.tfrecords')\n",
+        "The `tf.data` module also provides tools for reading and writing data in tensorflow."
+      ]
+    },
+    {
+      "metadata": {
+        "id": "1FISEuz8ubu3",
+        "colab_type": "text"
+      },
+      "cell_type": "markdown",
+      "source": [
+        "### Writing a TFRecord file\n",
         "\n",
-        "for i in range(n_observations):\n",
-        "  example = create_example([feature0[i], feature1[i], feature2[i], feature3[i]])\n",
-        "  writer.write(example.SerializeToString())\n",
+        "The easiest wayt to get the data into a dataset is to use the `from_tensor_slices` method.\n",
         "\n",
-        "writer.close()"
-      ],
-      "execution_count": 0,
-      "outputs": []
+        "Applied to an array, it returns a dataset of scalars."
+      ]
     },
     {
       "metadata": {
-        "id": "EjdFHHJMpUUo",
+        "id": "mXeaukvwu5_-",
         "colab_type": "code",
         "colab": {}
       },
       "cell_type": "code",
       "source": [
-        "!ls"
+        "tf.data.Dataset.from_tensor_slices(feature1)"
       ],
       "execution_count": 0,
       "outputs": []
     },
     {
       "metadata": {
-        "id": "wtQ7k0YWQ1cz",
+        "id": "f-q0VKyZvcad",
         "colab_type": "text"
       },
       "cell_type": "markdown",
       "source": [
-        "## Reading A `.tfrecords` File"
+        "Applies to a tuple of arrays, it returns a dataset of tuples:"
       ]
     },
     {
       "metadata": {
-        "id": "utkozytkQ-2K",
-        "colab_type": "text"
+        "id": "H5sWyu1kxnvg",
+        "colab_type": "code",
+        "colab": {}
       },
-      "cell_type": "markdown",
+      "cell_type": "code",
       "source": [
-        "Suppose we now want to read this data back, to be input as data into a model.\n",
-        "\n",
-        "The following example imports the data as is, as a `tf.Example` message. This can be useful to verify that a the file contains the data that we expect. This can also be useful if the input data is stored as TFRecords but you would prefer to input NumPy data (or some other input data type), for example [here](https://www.tensorflow.org/guide/datasets#consuming_numpy_arrays), since this example allows us to read the values themselves.\n",
-        "\n",
-        "We iterate through the TFRecords in the infile, extract the `tf.Example` message, and can read/store the values within."
-      ]
+        "features_dataset = tf.data.Dataset.from_tensor_slices((feature0, feature1, feature2, feature3))\n",
```

Comment:
did you try running this? in my example, I had to wrap `feature2` in `list(...)` for it to run correctly

Reply:
Yes. 

In the earlier version, one of the features was a single string instead of an array of strings.

I changed it to a list of strings where the feature is created.
The current version runs end-to-end.

You can try in the staging link:

https://colab.sandbox.google.com/github/MarkDaoust/docs/blob/tf-rec/site/en/tutorials/load_data/tf-records.ipynb